### PR TITLE
Update size recommendation endpoints

### PR DESCRIPTION
**Pull request by DharmeshBasapatiBacancy**
### Description

- Updated Size Recommendation API related URL, request model, response model.
- Added environment related size recommendation base URLs.
- Also did changes on test cases related to size recommendation API and verified it.
- Unit tested both the scenarios for male and female and their sizes.

### Demo

N/A

### Ticket

https://app.clickup.com/t/3702259/NSDK-178

### CODE REVIEW

File: virtusize-core/src/main/java/com/virtusize/android/network/VirtusizeApi.kt [hunk 1]
```diff
@@ -352,10 +353,11 @@ object VirtusizeApi {
     ): ApiRequest {
         val bodyProfileRecommendedSizeParams =
             BodyProfileRecommendedSizeParams(productTypes, storeProduct, userBodyProfile)
-        val url = Uri.parse(environment.servicesApiUrl() + VirtusizeEndpoint.GetSize.getPath())
-            .buildUpon()
-            .build()
-            .toString()
+        val url =
+            Uri.parse(environment.sizeRecommendationApiBaseUrl() + VirtusizeEndpoint.GetSize.getPath())
```
Comment (ourognao) [inline virtusize-core/src/main/java/com/virtusize/android/network/VirtusizeApi.kt:357]: Can't we use string templates ?
```kotlin
val url = "${environment.sizeRecommendationApiBaseUrl()}${VirtusizeEndpoint.GetSize.getPath()}"
```

**Review by ourognao — COMMENTED** [review]
@DharmeshBasapatiBacancy 1 comment

File: virtusize-core/src/main/java/com/virtusize/android/network/VirtusizeApi.kt (hunk 1 above)
Comment (DharmeshBasapatiBacancy) [reply virtusize-core/src/main/java/com/virtusize/android/network/VirtusizeApi.kt:357]: Yes we can do it.

**Review by ourognao — APPROVED** [review]
fat-face-cat

File: virtusize-core/src/main/java/com/virtusize/android/network/VirtusizeApi.kt (hunk 1 above)
Comment (ourognao) [reply virtusize-core/src/main/java/com/virtusize/android/network/VirtusizeApi.kt]: Verified thanks !

**Pull request merged**